### PR TITLE
Fix content body response on 304 code for #18199 issue

### PR DESCRIPTION
**Pull request by rad8329**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #18199

Fix content body response on 304 http status code, according to rfc7232

### CODE REVIEW

File: framework/web/Response.php [hunk 1]
```diff
@@ -1061,10 +1061,14 @@ protected function defaultFormatters()
...
      * The default implementation will convert [[data]] into [[content]] and set headers accordingly.
      * @throws InvalidConfigException if the formatter for the specified format is invalid or [[format]] is not supported
+     *
+     * @see https://tools.ietf.org/html/rfc7231#page-53
+     * @see https://tools.ietf.org/html/rfc7232#page-18
      */
     protected function prepare()
     {
-        if ($this->statusCode === 204) {
+        if ($this->statusCode === 204 || $this->statusCode === 304) {
```
Comment (samdark) [inline framework/web/Response.php:1070]: Isn't it the same for 100, 101, 102, 205?

Comment (rad8329) [reply framework/web/Response.php:1070]: The specification for 1xx does not say anything about of the content body response, it is not explicit like  rfc7231/rfc7232 are. By other hand the specification for 205 code status is more complex ([see here](https://tools.ietf.org/html/rfc7231#page-53)), maybe we need to prepare that response according to the rfc7231, but I think it is better to break it out in other issue.

Comment (samdark) [reply framework/web/Response.php:1070]: OK.

**Review by samdark — APPROVED** [review]

File: framework/web/Response.php (hunk 1 above)
Comment (githubjeka) [inline framework/web/Response.php:1070]: better use `$this->getStatusCode()`

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ Yii Framework 2 Change Log
 2.0.40 under development
 ------------------------
 
+- Bug #18199: Fix content body response on 304 http status code, according to rfc7232 (rad8329)
```
Comment (githubjeka) [inline framework/CHANGELOG.md:7]: 204 and 304

**Review by githubjeka — APPROVED** [review]

File: framework/web/Response.php (hunk 1 above)
Comment (rad8329) [reply framework/web/Response.php]: Also I've changed the condition expression, to reduce the cognitive load

**Review by githubjeka — APPROVED** [review]

**Review by bizley — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Comment by rad8329** [comment]
> Thank you!

It's a pleasure

**Pull request merged**